packages/nimbus/src/components/money-input/money-input.tsx

```diff
@@ -0,0 +1,373 @@
+      };
+      onValueChange?.(newValueObject);
+      onCurrencyChange?.(currencyCode as TCurrencyCode);
+    },
+    [onChange, onValueChange, onCurrencyChange, value, id, name]
+  );
+
+  const handleCurrencyFocus = useCallback(() => {
+    setCurrencyHasFocus(true);
+    const event: TCustomEvent = {
+      target: {
+        id: getCurrencyDropdownId(),
+        name: getCurrencyDropdownName(),
+        value: value.currencyCode,
+      },
+    };
+    onFocus?.(event);
+  }, [onFocus, value.currencyCode, id, name]);
+
+  const handleCurrencyBlur = useCallback(() => {
+    setCurrencyHasFocus(false);
+    const event: TCustomEvent = {
+      target: {
+        id: getCurrencyDropdownId(),
+        name: getCurrencyDropdownName(),
+        value: value.currencyCode,
+      },
+    };
+    onBlur?.(event);
+  }, [onBlur, value.currencyCode, id, name]);
+
+  const handleCurrencySelectionChange = useCallback(
+    (selectedKey: string | number | null) => {
+      if (selectedKey && typeof selectedKey === "string") {
+        handleCurrencyChange(selectedKey);
+      }
+    },
+    [handleCurrencyChange]
+  );
+
+  const stateProps = {
+    isInvalid,
+    isDisabled,
+    isReadOnly,
+    hasFocus,
+  };
+
+  return (
+    <MoneyInputRootSlot
+      {...recipeProps}
+      {...styleProps}
+      {...stateProps}
+      {...remainingProps}
+    >
+      <MoneyInputContainerSlot>
+        {/* Currency Select or Label */}
+        <MoneyInputCurrencySelectSlot {...stateProps}>
+          {hasNoCurrencies ? (
+            <MoneyInputCurrencyLabelSlot asChild>
+              <label data-testid="currency-label">{value.currencyCode}</label>
+            </MoneyInputCurrencyLabelSlot>
+          ) : (
+            <Select.Root
+              selectedKey={value.currencyCode || null}
+              onSelectionChange={handleCurrencySelectionChange}
+              onFocus={handleCurrencyFocus}
+              onBlur={handleCurrencyBlur}
+              isDisabled={isCurrencyInputDisabled || isDisabled || isReadOnly}
+              isClearable={false}
+              placeholder=""
+              aria-label={intl.formatMessage(messages.currencySelection)}
+              data-testid="currency-dropdown"
+              size={size}
+            >
+              <Select.Options>
+                {currencies.map((currencyCode) => (
+                  <Select.Option key={currencyCode} id={currencyCode}>
+                    {currencyCode}
+                  </Select.Option>
+                ))}
+              </Select.Options>
+            </Select.Root>
+          )}
+        </MoneyInputCurrencySelectSlot>
+
+        {/* Amount Input - NumberInput handles live user interaction */}
+        <MoneyInputAmountInputSlot
+          ref={ref}
+          data-has-focus={hasFocus}
+          data-has-high-precision={
+            hasHighPrecisionBadge && isCurrentlyHighPrecision
+          }
+          {...stateProps}
+          asChild
+        >
+          <NumberInput
```

Comment:
I would try this for handling the display of high precision currency, based on [the docs for Intl.NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#maximumfractiondigits):
```suggestion
          <NumberInput
            formatOptions={{ 
              maximumFractionDigits: isCurrentlyHighPrecision ? 20 : undefined
             }}
```